src/main/kotlin/ai/whylabs/services/whylogs/util/CoroutineUtil.kt

```diff
@@ -0,0 +1,22 @@
+package ai.whylabs.services.whylogs.util
+
+import kotlinx.coroutines.CancellationException
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.isActive
+import kotlinx.coroutines.yield
+import org.slf4j.Logger
+
+suspend inline fun CoroutineScope.repeatUntilCancelled(log: Logger, block: () -> Unit) {
+    while (isActive) {
+        try {
+            block()
+            yield()
+        } catch (ex: CancellationException) {

```

Comment:
It actually did. Added a throw to that and explicitly allowed the `TimeoutCancellationException` to not kill the loop since those are thrown for timeouts in the handler.